pkg/asset/installconfig/vsphere/validation.go

```diff
@@ -66,3 +82,99 @@ func folderExists(ic *types.InstallConfig, fldPath *field.Path) field.ErrorList
 	}
 	return nil
 }
+
+func validatevCenterVersion(vim25 *vim25.Client) error {
+	constraints, err := version.NewConstraint("<= 6.7")
+
+	if err != nil {
+		return err
+	}
+
+	vCenterVersion, err := version.NewVersion(vim25.ServiceContent.About.Version)
+	if err != nil {
+		return err
+	}
+	build, err := strconv.Atoi(vim25.ServiceContent.About.Build)
+	if err != nil {
+		return err
+	}
+
+	if constraints.Check(vCenterVersion) {
+		logrus.Warnf("The out-of-tree storage driver requires virtual hardware 15 and vSphere 6.7 U3. Current vCenter version: %s, build: %s",
+			vim25.ServiceContent.About.Version, vim25.ServiceContent.About.Build)
+
+	} else {
+		// This is the vCenter 6.7 U3 build number
+		// Anything less than this version is unsupported with the
+		// out-of-tree CSI.
+		// https://kb.vmware.com/s/article/2143838
+		// https://vsphere-csi-driver.sigs.k8s.io/supported_features_matrix.html
+		if build < 14368073 {
```

Comment:
similar to the comment from the other PR, would be good to have this variable as a constant.

Reply:
if possible I would like us to be consistent with how upstream reports similar warnings for using older versions of vcenter - https://github.com/kubernetes/kubernetes/pull/98546/files#diff-494c25f28f8cd90f6fe6fa4a32a6f59e457c34bc6bbdb992d4c75d7f6138e683R205

We have to carry similar code in vsphere-problem-detector and we should be consistent everywhere in checks we perform.